hudi-common/src/main/java/org/apache/hudi/common/engine/HoodieReaderContext.java

```diff
@@ -239,10 +250,53 @@ public abstract ClosableIterator<T> getFileRecordIterator(
    * @param storage         {@link HoodieStorage} for reading records.
    * @return {@link ClosableIterator<T>} that can return all records through iteration.
    */
-  public ClosableIterator<T> getFileRecordIterator(
-      StoragePathInfo storagePathInfo, long start, long length, Schema dataSchema, Schema requiredSchema,
-      HoodieStorage storage) throws IOException {
-    return getFileRecordIterator(storagePathInfo.getPath(), start, length, dataSchema, requiredSchema, storage);
+  public final ClosableIterator<T> getFileRecordIterator(
```

Comment:
nit: find a way to reduce the number of methods getting record iterators, i.e., `getFileRecordIterator`, `doGetFileRecordIterator`, `getFileRecordIteratorInternal`.

Reply:
+1